src/secrets/components/SecretCard.tsx

```diff
@@ -25,32 +22,32 @@ interface Props {
 const SecretCard: FC<Props> = ({secret}) => {
   const dispatch = useDispatch()
 
-  const handleCopyAttempt = (
-    copiedText: string,
-    isSuccessful: boolean
-  ): void => {
-    if (isSuccessful) {
-      dispatch(notify(copyToClipboardSuccess(copiedText, 'Secret ID')))
-    } else {
-      dispatch(notify(copyToClipboardFailed(copiedText, 'Secret ID')))
-    }
+  const handleDelete = () => {
+    dispatch(deleteSecret(secret))
+  }
+
+  const history = useHistory()
+  const orgId = useSelector(getOrg)?.id
+
+  const editSecret = () => {
+    history.push(`/orgs/${orgId}/settings/secrets/${secret.key}/edit`)
   }
 
   return (
     <ErrorBoundary>
       <ResourceCard
-        key={`secret-id--${secret.id}`}
-        testID={`secret-card--${secret.id}`}
-        contextMenu={<SecretContextMenu secret={secret} />}
+        testID={`secret-card--${secret?.id}`}
+        contextMenu={
+          <SecretContextMenu secret={secret} onDeleteSecret={handleDelete} />
+        }
         direction={FlexDirection.Row}
         alignItems={AlignItems.Center}
       >
-        <CopyToClipboard text={secret.id} onCopy={handleCopyAttempt}>
-          <ResourceCard.Name
-            name={secret.id}
-            testID={`secret-card--name ${secret.id}`}
-          />
-        </CopyToClipboard>
+        <ResourceCard.Name
+          name={secret?.id}
+          testID={`secret-card--name-${secret?.id}`}
```

Comment:
I remember seeing id references changed to `key`, should we do that here?

Reply:
Yep.  Changed.